src/shared/utils/inner-fetch.ts

```diff
@@ -0,0 +1,25 @@
+import { deserialize, Serialized } from './serializer';
+
+/**
+ * A wrapper around `fetch` to request data from local endpoints. Features:
+ * 1. Composes the URL search params correctly, only provide an object
+ * 2. Throws if the response contains an error object
+ * 3. Deserializes the response object, if it contains protobuf messages
+ */
+export const innerFetch = async <RES extends object>(
```

Comment:
suggestion: `inner` feels a bit like a private field. I wonder if `apiFetch` or `fetchWithSerialization` or `structuredFetch` may be better.

Reply:
thanks, `apiFetch` feels like a better name